src/app/controllers/VendorController.ts

```diff
@@ -155,89 +141,32 @@ export interface OpenHoursInput {
   sunday: Array<TimeSpanInput>;
 }
 
-export interface OpenHours {
-  monday?: Array<TimeSpanInput>;
-  tuesday?: Array<TimeSpanInput>;
-  wednesday?: Array<TimeSpanInput>;
-  thursday?: Array<TimeSpanInput>;
-  friday?: Array<TimeSpanInput>;
-  saturday?: Array<TimeSpanInput>;
-  sunday?: Array<TimeSpanInput>;
+export interface LocationInput extends Location {
+  
 }
 
-export interface LocationInput {
+export interface Location {
   longitude: number;
   latitude: number;
 }
 
-export interface Location {
-  longitude?: number;
-  latitude?: number;
+export interface DirectDepositInfoInput extends DirectDepositInfo {
+  
 }
 
-export interface DirectDepositInfoInput {
+export interface DirectDepositInfo  {
   transit_no: string;
   institution_no: string;
   account_no: string;
   cheque_image: string;
 }
 
-export interface DirectDepositInfo {
-  transit_no?: string;
-  institution_no?: string;
-  account_no?: string;
-  cheque_image?: string;
-}
-
-export interface VendorCommonProperties {
-  name: string;
-  description: string;
-  images: Array<string>;
-  address: string;
-  phone_number: string;
-  status: VendorStatus;
-  payout_email_address: string;
-  directions: string;
-  receive_sms_notifications: boolean;
-  auto_open: boolean;
-  auto_close: boolean;
-  payout_auto_request_schedule: PayoutAutoRequestSchedule;
-  global_tax_rate: number;
-  postal_code: string;
-  store_logo: string;
-  max_group_seating_limit: number;
-}
-
-export interface UpdateVendorInput {
-  tags?: Array<TagInput>;
-  open_hours?: OpenHoursInput;
-  location?: LocationInput;
-  direct_deposit_info?: DirectDepositInfoInput;
-  payment_methods?: PaymentMethodsInput;
-  order_types?: OrderTypesInput;
-  name?: string;
-  description?: string;
-  images?: Array<string>;
-  address?: string;
-  phone_number?: string;
-  status?: VendorStatus;
-  payout_email_address?: string;
-  directions?: string;
-  receive_sms_notifications?: boolean;
-  auto_open?: boolean;
-  auto_close?: boolean;
-  payout_auto_request_schedule?: PayoutAutoRequestSchedule;
-  global_tax_rate?: number;
-  postal_code?: string;
-  store_logo?: string;
-  max_group_seating_limit?: string;
-  website_link?: string;
-}
+export interface UpdateVendorInput extends Pick<Vendor, 'name' | 'description' | 'images' | 'tags' | 'open_hours' | 'address' | 'phone_number' | 'location' | 'courier_pickup_instructions' | 'status' | 'payout_email_address' | 'direct_deposit_info' | 'payment_methods' | 'directions' | 'receive_sms_notifications' | 'auto_open' | 'auto_close' | 'payout_auto_request_schedule' | 'global_tax_rate' | 'postal_code' | 'order_types' | 'store_logo' | 'min_delivery_cart_total' | 'max_group_seating_limit' | 'website_link' > {
```

Comment:
Why not do Omit instead of all this haha? Also where is the Partial??

Reply:
I know.. i try omit too. The numbers of keys are huge. with Pick<> become shorter.